### PR TITLE
AADT and K_factor

### PR DESCRIPTION
**Pull request by mmmiah**
This PR calculated 8 different AADT at the station level that is described details in issue #185. It also calculates K-factors for each station. Please have your review and point out if there is any missing logic. Thanks!

### CODE REVIEW

File: transform/models/intermediate/clearinghouse/_clearinghouse.yml [hunk 1]
```diff
@@ -474,3 +474,67 @@ models:
           The lost Productivity perfromance metric is
           the number of lane-mile-month on the freeway lost due to operating under congested
           conditions instead of under free-flow conditions.
+  - name: int_clearinghouse__annual_average_daily_traffic
```
Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/_clearinghouse.yml:477]: Per our conventions discussed in #241, I think this might be classified as a "performance-related" model, and be named something like `int_performance__station_annual_average_daily_traffic`. I think it's complex enough that it doesn't really need to fit into the specific naming convention for temporal aggregates, however

File: transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql
Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql:1]: Can you add YAML for this model?

Similar to the above, I think this might be more appropriate in the "performance" directory

File: transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        id,
+        district,
+        type,
+        observation_month,
+        kfactor_month,
+        hourly_volume as k_factor
+    from
+        traffic_with_rank
+    where
+        traffic_rank = 30
```
Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql:40]: In #242 this is described as 

>  Along with the 8 AADT values, PeMS computes a "k-factor" for each station and for each month. This is defined to be the 30th highest hourly traffic flow observed at that station among all hours of the preceding year.

Is that description wrong? This computes the k-factor for the preceding month, but the description says for the preceding year.

Comment (ian-r-rose) [inline transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql:40]: There's an opportunity to consolidate these last two CTEs by using a [`qualify`](https://docs.snowflake.com/en/sql-reference/constructs/qualify) clause in `traffic_with_rank`:

```sql
qualify RANK() over (
    partition by id, observation_month
    order by hourly_volume desc
) = 30
```

**Review by ian-r-rose — COMMENTED** [review]
This is really great work @mmmiah! There *may* be a way to combine a few of the different metrics and reduce the join count, but to be honest that's probably not worth any savings at that level of aggregation. The SQL looks well-structured and readable to me, though I don't really understand the motivation behind the metrics as a non-traffic engineer.

File: transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql (hunk 2 above)
Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql]: Yes, I think that it will be preceding years only, not the month, A single value in a particular year for a station
image

Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql]: @ian-r-rose ,Great suggestion! I just incorporated your suggestion. Thank you!

Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql]: I have also included K-50 and K_100, but I think that we are not going to get the factor as we have 8 AADT, We will leave the option for users to choose the AADT to get the final k_factors. We will just report 30th, 50th and 100th highest hourly volume in a particular year. We will discuss this things with PeMS committee also.

File: transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql
Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/int_clearninghouse__station_k_factor.sql:1]: I have combined k factor model with AADT model and also included ymal for this model under performance. Thanks for your review. Appreciate!

File: transform/models/intermediate/clearinghouse/_clearinghouse.yml (hunk 1 above)
Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/_clearinghouse.yml]: @ian-r-rose ,Yes, named as 'int_performance__station_annual_average_daily_traffic' We have also k value in this model. how about naming it as ''int_performance__station_annual_average_daily_traffic_with_k_value (not exactly factor as we did not divide it with AADT, we just reported 30th, 50th and 100 th highest hour volume). Let me know if you have something better in mind

**Comment by mmmiah** [comment]
This PR read for final review and merge! Thank you!

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        district,
+        direction,
+        freeway,
+        type,
+        sample_year,
+        count(sample_month) as sample_ct,
+        avg(madt) as aadt_2
+    from madt
+    group by id, city, county, district, freeway, direction, type, sample_year
+    having count(id) >= 12
```
Comment (ian-r-rose) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:69]: Can you expand on why we need this `having` statement? Is 12 corresponding to the number of months? Why would we expect there to be more than 12, if the previous aggregate is per-month?

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+        county,
+        district,
+        freeway,
+        direction,
+        type,
+        sample_year,
+        avg(aadw) as aadt_3
+    from aadw
+    group by id, city, county, district, freeway, direction, type, sample_year
+    having count(id) = 7
```
Comment (ian-r-rose) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:103]: How often do we get a station with a different number than 7? Does that happen if, e.g., a station is only operational for a few days out of the year, and doesn't have samples for every day of the week?

**Review by ian-r-rose — APPROVED** [review]
This looks great to me @mmmiah! My only suggestion is that it might be helpful to add a few comments on your `having` statements to make it clearer what they are filtering out. But I see that it's well documented in the YAML, so I view the suggestion as optional

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 3 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql]: Yes, none of madt should be missing in 12 months for this aadt version. It should not be more than 12. I just replaced it as '=12' and also included the notes as 
    -- all 12 months average traffic volume is required to calculate this metric
    --other way, there should be no missing monthly average daily traffic

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 4 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:106]: I have done sample count for this, it seems that data is not really complete for many cases. That's why we have 2nd version of this aadt which is named as aadt_7. 
AADT_7: Modified Conventional AASHTO
This procedure is a modified version of AADT_3, in which 1 of the 12 MADW values may be missing in the AADW subcomputation, and/or 1 of the 7 AADW values may be missing in the final AADT_7 computation. 

I have included the clarification notes for having count(id)=7.

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql
Comment (ZhenyuZhu-Caltrans) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:1]: for aadt2: 
having count(id) >= 12 ----> having count(sample_month) >=12

for aadt3:
I did not see the requirement of all average daily traffic of 7 days of the week needs to be nun-null to calculate the AADT. 
delete having count(id) = 7

for aadt4:
I did not see the requirement of all average daily traffic of 7 days of the week needs to be nun-null to calculate the AADT. 
delete having count(id) = 7

for aadt7:
I do not see the code for 
in which 1 of the 12 MADW values may be missing in the AADW subcomputation,

for aadt8:
I do not see the code for 
in which 1 of the 12 MADW values may be missing in the AADW subcomputation,

File: transform/models/intermediate/imputation/_imputation.yml [hunk 5]
```diff
@@ -1,77 +1,77 @@
...
+          tagging wheather the other id is local counters
+          or regional counters. If the other id is just upstream or downstream
+          then it is considered as local counters. On the other hand
+          except local counters if any counters are located within 5 miles
+          upstream and downstream, then it is considered as regional id.
+          This boolean function where True means local counters and false means
+          regional counters.
+  - name: int_imputation__detector_agg_five_minutes
+    description: |
+      dfdf
```
Comment (britt-allen) [inline transform/models/intermediate/imputation/_imputation.yml:50]: This is incomplete

File: transform/models/intermediate/performance/_performance.yml [hunk 6]
```diff
@@ -343,3 +343,73 @@ models:
           The lost Productivity perfromance metric is
           the number of lane-mile-month on the freeway lost due to operating under congested
           conditions instead of under free-flow conditions.
+  - name: int_performance__station_aadt_with_K_value
+    description: |
+      Station level annual average daily traffic and k value
```
Comment (britt-allen) [inline transform/models/intermediate/performance/_performance.yml:348]: What is k value?

File: transform/models/intermediate/clearinghouse/_clearinghouse.yml
Comment (britt-allen) [inline transform/models/intermediate/clearinghouse/_clearinghouse.yml:1]: I would add periods to the ends of all sentences. Some have them and others do not so it's a bit inconsistent.

File: transform/models/intermediate/performance/_performance.yml
Comment (britt-allen) [inline transform/models/intermediate/performance/_performance.yml:1]: Same here. I would add periods to the ends of all sentences. Some have them and others do not so it's a bit inconsistent.

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
+        avg(daily_volume) as aadt_1,
+        date_trunc('year', sample_date) as sample_year
+    from {{ ref('int_performance__station_metrics_agg_daily') }}
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:15]: you select from this table 2 times, the best practice it to put this in a CTE selecting all or only the columns you need. for example, 

`with daily_station_metrics as (
     select * from {{ ref('int_performance__station_metrics_agg_daily') }}
)`

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+-- AADT_2: ASTM Standard 1442
+madw as (
+    select
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:28]: You can remove the 7 lines of code that it takes to individually list these columns (id, city, county, district, freeway, direction, type,) and replace it with a *

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+        id,
+        district,
+        type,
+        extract(dow from sample_date) as day_of_week,
+        extract(hour from sample_hour) as hour_of_day,
+        date_trunc('month', sample_date) as sample_month,
+        date_trunc('year', sample_date) as sample_year,
+        -- calculate monthly average flow for each hour of the week
+        avg(hourly_volume) as mahw
+    from {{ ref('int_performance__station_metrics_agg_hourly') }}
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:121]: you select from this table 3 times, the best practice it to put this in a CTE selecting all or only the columns you need. for example, 

`with hourly_station_metrics as (
     select * from {{ ref('int_performance__station_metrics_agg_hourly') }}
)`

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+    select
+        id,
+        city,
+        county,
+        district,
+        direction,
+        freeway,
+        type,
+        sample_month,
+        sample_year,
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:48]: same here, you can remove the 9 lines of code that it takes to individually list these columns (id, city, county, district, freeway, direction, type, sample_month, sample_year,) and replace it with a *

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+aadt_2 as (
+    select
+        id,
+        city,
+        county,
+        district,
+        direction,
+        freeway,
+        type,
+        sample_year,
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:64]: same here, you can select *

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+-- AADT_3: Conventional AASHTO Procedures
+aadw as (
+    select
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:83]: same here, you can select *

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 13]
```diff
@@ -0,0 +1,362 @@
...
+aadt_3 as (
+    select
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
+        sample_year,
```
Comment (britt-allen) [inline transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:101]: same here, you can select *

**Review by britt-allen — COMMENTED** [review]
This is looking good so far! Right now the file is extremely long and hard to parse so I recommend selecting star where you can. I also recommend making use of reusable CTEs rather than directly selecting from the same table multiple times. Even if you select from the table once it's a best practice to place your source data in a CTE.

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 13 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:101]: It pops as ambiguous error if I do not choose the specific column before aggregating

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 12 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:83]: It pops as ambiguous error if I do not choose the specific column before aggregating

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 11 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:64]: It pops as ambiguous error if I do not choose the specific column before aggregating

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 10 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:48]: It pops as ambiguous error if I do not choose the specific column before aggregating

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 9 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:121]: dropped it and called from very first read

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 7 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:15]: read once the daily and hourly file and then call the necessary column that I needed

File: transform/models/intermediate/performance/_performance.yml
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:1]: added period for all ymal description

File: transform/models/intermediate/clearinghouse/_clearinghouse.yml
Comment (mmmiah) [reply transform/models/intermediate/clearinghouse/_clearinghouse.yml:1]: added the period at the end of all sentences

File: transform/models/intermediate/performance/_performance.yml (hunk 6 above)
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:348]: K Value is defined at the end of this model ymal as 
      - name: K_30
        description: 30th highest hour volume in the precceding year
      - name: K_50
        description: 50th highest hour volume in the precceding year
      - name: K_100
        description: 100th highest hour volume in the precceding year

Also edited the model description as:   - name: int_performance__station_aadt_with_K_value
    description: |
      Station level annual average daily traffic and k th highest hour volume in the precceding year

File: transform/models/intermediate/imputation/_imputation.yml (hunk 5 above)
Comment (mmmiah) [reply transform/models/intermediate/imputation/_imputation.yml:50]: Let me do that in imputation branch and keep this separate for now

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:1]: Changed-
for aadt2:
having count(id) >= 12 ----> having count(sample_month) =12
for aadt3:
Very less clear description. I have read through other resources and found there is no requirements for this restriction
I have deleted 'having count(id) = 7'

For aadt_4:
 Dropped the condition 'having count(id) = 7'
for aadt7:
I thought 'or' indicates just to apply one condition. As it is said and/or, based on your suggestions, I have incorporated both condition as 
    -- 1 of the 12 MADW values may be missing in the AADW subcomputation.
    having count(madw) >= 11  
    -- 1 of the 7 aadw values may be missing for this final AADT calculation.
    having count(aadw) >= 6
for aadt8:
Applied both condition now as 
    -- 1 of the 12 MADW values may be missing in the AADW subcomputation
    having count(madw) >= 11
    -- 1 of the 7 aadw values may be missing for this final AADT calculation.
    having count(aadw) >= 6

**Comment by mmmiah** [comment]
@ZhenyuZhu-Caltrans @britt-allen 
Can you have 2nd look for AADT and k factor model? I am planning to merge this in main before Wednesday sprint.

**Review by kengodleskidot — COMMENTED** [review]
Looks great @mmmiah just some high-level comments that you can address before or after this PR.

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+
+monthly_average_days_of_the_week_traffic as (
+    select
+        id,
+        district,
+        type,
+        day_of_week,
+        sample_month,
+        sample_year,
+        sum(mahw) as madw
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:143]: Applied additional filter to drop zero value of madw for aadt calculation. However, we will not filter out hourly zeros as at late night like 3 pm may be zero for the whole week that will not surprise me. However, you are right that monthly average day of week traffic zero does not make sense. I just applied additional filters to avoid zero skewed concern.

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 15]
```diff
@@ -0,0 +1,407 @@
...
+    select
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
+        -- calculate  Monthly Average Days of the Week (MADW).
+        avg(daily_volume) as madw,
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:36]: The zero values of madw was not filtered previously that may potentially create null values. I have added additional filter for madw value in all CTEs

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 16]
```diff
@@ -0,0 +1,407 @@
...
+        direction,
+        freeway,
+        type,
+        sample_year,
+        count(sample_month) as sample_ct,
+        avg(madt) as aadt_2
+    from madt
+    group by id, city, county, district, freeway, direction, type, sample_year
+    -- all 12 months average traffic volume is required to calculate this metric
+    -- otherway,there should be no missing monthly average daily traffic
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:76]: first line comments seems sufficient. Just dropped the 2nd line comment to avoid confusion

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql [hunk 17]
```diff
@@ -0,0 +1,407 @@
...
+aadt_1 as (
+    select
+        id,
+        city,
+        county,
+        district,
+        freeway,
+        direction,
+        type,
+        avg(daily_volume) as aadt_1,
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:19]: If you see that some CTEs initially have only id, district, type, these CTEs specially depending on station hourly agg file where we do not have all details meta info due to expensive joining issues (previously discussed). We do not have those details in hourly model to have them here.

File: transform/models/intermediate/performance/_performance.yml [hunk 18]
```diff
@@ -343,6 +343,76 @@ models:
...
+        description: |
+        Similarly, this procedure is a modified version of AADT_4, in which 1 of the 12 MADW
+        values may be missing in the AADW subcomputation, and/or 1 of the 7 AADW values may
+        be missing in the final AADT_8 computation. This is Modified Provisional AASHTO.
+      - name: K_30
+        description: 30th highest hour volume in the precceding year.
+      - name: K_50
+        description: 50th highest hour volume in the precceding year.
+      - name: K_100
+        description: 100th highest hour volume in the precceding year.
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:415]: Yes, K-Values are based on observed data only, I think that I need to filter the data for good detectors only that is not implemented in current version. I believe that it is better to filter the good detectors in spatial temporal aggs as this model depend on previous aggregation. What do you think?

File: transform/models/intermediate/performance/_performance.yml [hunk 19]
```diff
@@ -343,6 +343,76 @@ models:
           The lost Productivity perfromance metric is
           the number of lane-mile-month on the freeway lost due to operating under congested
           conditions instead of under free-flow conditions.
+  - name: int_performance__station_aadt_with_K_value
+    description: |
+      Station level annual average daily traffic and k th highest hour volume in the precceding year.
```
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:348]: Included the additional resource link in the ymal file. Currently, there is no imputed data used for this AADT calculation as all calculation depends on source file 'int_performance__detector_metrics_agg_five_minutes'. We did not filter out bad detectors in this PR. How about filtering the good detectors in spatial temporal aggregation first, then our this model should be as it is?

[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:348]: @kengodleskidot based on our discussion, I have filtered out the good detectors first, then aggregated volume in station level to further calculate the AADT and k value!

File: transform/models/intermediate/performance/_performance.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/_performance.yml:420]: Data filtered for good detectors only based on discussion

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 15 above)
[the remark this answers is not in the document]
Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:70]: Some of the aadt (like aadt_1,6,7 and 8) still may be null due to not having enough sample to satisfy the condition such as 
having count(sample_month) = 12 for aadt_2
 having count(madt) >= 11 for aadt_6
having count(madw) >= 11 and having count(aadw) >= 6 for aadt_7,
having count(madw) >= 11 and having count(aadw) >= 6 for aadt_8

**Comment by mmmiah** [comment]
@kengodleskidot , ready for your re-review!

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 10 above)
Comment (britt-allen) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:88]: can you share a screenshot of the error? i'm not familiar

Comment (mmmiah) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:88]: After last edit and revision, right now under each CTE there is some calculation needed, therefore, I believe that we can not simply select everything using *.  If you select everything from the CTE and then join or aggregate sometimes it says ambiguous error. Will share such things when face again for that specific case

File: transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql (hunk 14 above)
[the remark this answers is not in the document]
Comment (kengodleskidot) [reply transform/models/intermediate/performance/int_performance__station_aadt_with_K_value.sql:197]: Thank you @mmmiah! I agree that the hourly data can contain zero values and your example is a good one. My main concern is if we are seeing daily volume values of 0 along the ML and HV stations which I do not believe should be the case unless all lanes for a station are impacted by an incident, construction or some other external issue. Regardless I do not believe we would want to use a daily volume value of 0 in the AADT calculation.

**Review by kengodleskidot — APPROVED** [review]
Looks great @mmmiah! Something that may warrant additional discussion is how the AADT values are affected when some samples are missing, e.g. 110 samples received in an hour instead of 120 expected.

**Comment by ian-r-rose** [comment]
Wooo! Congrats @mmmiah, and thanks for the detailed review @kengodleskidot!
